src/crypto/transform.rs

```diff
@@ -18,26 +18,36 @@ pub fn generate_new_doc_key<CR: rand::CryptoRng + rand::RngCore>(
 }
 
 /// Generate a plaintext and a keypair necessary to create a new group
-pub fn gen_group_keys<CR: rand::CryptoRng + rand::RngCore>(
-    recrypt: &Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
+pub fn gen_group_keys<R: CryptoOps + KeyGenOps>(
+    recrypt: &R,
 ) -> Result<(Plaintext, PrivateKey, PublicKey), IronOxideErr> {
     let plaintext = recrypt.gen_plaintext();
     let priv_key = recrypt.derive_private_key(&plaintext);
     let pub_key = recrypt.compute_public_key(&priv_key)?;
-
     Ok((plaintext, priv_key.into(), pub_key.into()))
 }
 
-/// Decrypt the provided encrypted plaintext and return both the plaintext and the symmetric key that
+/// Decrypt the provided encrypted plaintext and return the symmetric key that is derived from it.
+pub fn decrypt_symmetric_key<CR: rand::CryptoRng + rand::RngCore>(
```

Comment:
suggestion: `decrypt_as_symmetric_key`

Reply:
Changed. I also changed the companion function to `decrypt_as_private_key` to match.